### PR TITLE
WRN-4404: Migrate Input and Keypad unit tests to Testing Library

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Enact-DCO-1.0-Signed-off-by: Stanca Pop stanca.pop@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [x] Documentation was verified or is not changed
* [x] UI test was passed or is not needed
* [x] Screenshot test was verified or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Migrated unit tests from Enzyme to Testing Library for Input and Keypad components.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRN-4404

### CODE REVIEW

**Review by alexandrumorariu — APPROVED** [review]
LGTM

File: Input/tests/Input-specs.js [hunk 1]
```diff
@@ -353,69 +297,72 @@ describe('Input Specs', () => {
...
-		clearButton.simulate('click');
 		const expectedValue = '';
 
-		setTimeout(() => expect(actualValue).toBe(expectedValue), 100); // setTimeout is used here to give input some time to clear its value
+		await waitFor(() => {
+			const actual = handleChange.mock.calls[0][0].value;
+
+			expect(actual).toBe(expectedValue);
+		});
+		// waitFor is used here to give input some time to call onChange
```
Comment (adrian-cocoara-lgp) [inline Input/tests/Input-specs.js:348]: maybe move this comment above the waitFor keyword

File: Input/tests/Input-specs.js [hunk 2]
```diff
@@ -353,69 +297,72 @@ describe('Input Specs', () => {
...
 
-		clearButton.simulate('click');
-		const expectedValue = value;
+		await waitFor(() => {
+			const actual = handleChange.mock.calls.length;
 
-		setTimeout(() => expect(actualValue).toBe(expectedValue), 100); // setTimeout is used here to give input some time to clear its value
+			expect(actual).toBe(expectedValue);
+		});
+		// waitFor is used here to give input some time to call onChange
```
Comment (adrian-cocoara-lgp) [inline Input/tests/Input-specs.js:366]: same here, move the comment above the waitFor keyword

File: Keypad/tests/Keypad-specs.js [hunk 3]
```diff
@@ -44,13 +37,11 @@ describe('Keypad Specs', () => {
...
 
-			click(keypad, 2);
-			click(keypad, 5);
-			click(keypad, 7);
+			userEvent.click(screen.getByLabelText('2'));
+			userEvent.click(screen.getByLabelText('5'));
+			userEvent.click(screen.getByLabelText('7'));
 
 			const expected = 0;
 			const actual = handleChange.mock.calls.length;
```
Comment (adrian-cocoara-lgp) [inline Keypad/tests/Keypad-specs.js:47]:
```suggestion
			test('should not run the onChange handler when disabled', () => {
			const handleChange = jest.fn();
			render(<Keypad disabled onChange={handleChange} />);

			userEvent.click(screen.getByLabelText('2'));
			userEvent.click(screen.getByLabelText('5'));
			userEvent.click(screen.getByLabelText('7'));

			expect(handleChange).not.toHaveBeenCalled();
		});
```

**Pull request merged**